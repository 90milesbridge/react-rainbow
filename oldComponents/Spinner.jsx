import React from 'react';
import classnames from 'classnames';

export default class Spinner extends React.Component {
    render() {
        let computedClasses = classnames('slds-spinner', {
            'slds-spinner--brand': this.props.variant === 'brand',
            'slds-spinner--inverse': this.props.variant === 'inverse',
            'slds-spinner--small' : this.props.size === 'small',
            'slds-spinner--medium' : this.props.size === 'medium',
            'slds-spinner--large' : this.props.size === 'large'
        });
        return this.props.isVisible ? (
            <div className="slds-spinner_container" onWheel={ e => e.preventDefault() }>
                <div className={ computedClasses } role="alert">
                    <span className="slds-assistive-text">Loading</span>
                    <div className="slds-spinner__dot-a"></div>
                    <div className="slds-spinner__dot-b"></div>
                </div>
            </div>
        ) : null;
    }
}

Spinner.PropTypes = {
    variant: React.PropTypes.string,
    size: React.PropTypes.string,
    isVisible: React.PropTypes.bool.isRequired
};

Spinner.defaultProps = {
    size: 'medium'
};