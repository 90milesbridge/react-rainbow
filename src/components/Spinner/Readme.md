List loading using Spinner - Base - Medium:

     const containerStyles = {
        position: 'relative',
        padding: 20,
    };

    <div className="slds-card-wrapper">
        <div style={containerStyles}>
            <Spinner size="medium" />
        </div>
    </div>

Full screen loading using Spinner - Brand - Large:

     const cardContainerStyles = {
            height: 268,
    };

    const spinnerTextStyles = {
            color: '#0070d2',
    };

    <Card className="slds-align_absolute-center slds-grid--vertical" style={cardContainerStyles} >
        <div className="slds-is-relative">
            <Spinner variant="brand" size="large" />
        </div>
        <h1 className="slds-text-heading_small slds-m-top_xx-large" style={spinnerTextStyles}>Loading…</h1>
    </Card>

Lazy loading using Spinner - Base - Small

        const spinner = (
            <div className="slds-align_absolute-center">
                <div className="slds-is-relative">
                    <Spinner size="small" />
                </div>
                <h1 className="slds-text-heading_small slds-m-left_large" style={{color: '#2a426c'}}>Loading…</h1>
            </div>
        );

        const childrenStyles = {
            height: 150,
    };

        <div className="slds-card-wrapper">
            <Card
                className="slds-card_boundary"
                iconName="standard:account"
                title="Accounts (6)"
                footer={spinner}
                actions={<Button variant="neutral" label="New" />} >
                    <div style={childrenStyles} />
            </Card>
        </div>
        
