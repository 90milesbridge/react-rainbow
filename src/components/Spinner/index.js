import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import AssistiveText from './../AssistiveText';

/**
* Spinners should be shown when retrieving data or performing slow computations,
* help to reassure the user that the system is actively retrieving data.
*/
export default function Spinner(props) {
    const {
        className,
        style,
        assistiveText,
        isVisible,
        size,
        variant,
    } = props;
    const getVariantClassNames = () => {
        if (variant === 'base') {
            return '';
        }
        return `slds-spinner_${variant}`;
    };

    const getContainerClassNames = () => classnames(
        'slds-spinner',
        `slds-spinner_${size}`,
        getVariantClassNames(),
        className,
    );

    if (isVisible) {
        return (
            <div className={getContainerClassNames()} style={style} role="status">
                <AssistiveText text={assistiveText} />
                <div className="slds-spinner__dot-a" />
                <div className="slds-spinner__dot-b" />
            </div>
        );
    }
    return null;
}

Spinner.propTypes = {
    /** The variant changes the appearance of the spinner.
    * Accepted variants are base, brand, and inverse. This value defaults to base. */
    variant: PropTypes.oneOf([
        'base', 'brand', 'inverse',
    ]),
    /** The size of the spinner. Accepted sizes are small, medium, and large.
    * This value defaults to medium. */
    size: PropTypes.oneOf([
        'xx-small',
        'x-small',
        'small',
        'medium',
        'large',
    ]),
    /** Show/Hide the spinner. */
    isVisible: PropTypes.bool,
    /** A description for assistive sreen readers. */
    assistiveText: PropTypes.string,
    /** A CSS class for the outer element, in addition to the component's base classes. */
    className: PropTypes.string,
    /** An object with custom style applied to the outer element. */
    style: PropTypes.object,
};

Spinner.defaultProps = {
    variant: 'base',
    size: 'medium',
    isVisible: true,
    assistiveText: null,
    className: undefined,
    style: undefined,
};
