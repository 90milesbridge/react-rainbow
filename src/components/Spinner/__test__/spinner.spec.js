import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Spinner from './../index';

describe('<Spinner/>', () => {
    it('should not render the spinner when isVisible is false', () => {
        const component = mount(
            <Spinner isVisible={false} />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should not render the spinner when isVisible is true', () => {
        const component = mount(
            <Spinner isVisible />,
        );
        expect(component.children().length).toBe(1);
    });
    it('should not render the assistive text when is not passed', () => {
        const component = mount(
            <Spinner />,
        );
        expect(component.find('.slds-assistive-text').length).toBe(0);
    });
    it('should render the assistive text passed', () => {
        const component = mount(
            <Spinner assistiveText="for screen readers" />,
        );
        expect(component.find('.slds-assistive-text').text()).toBe('for screen readers');
    });
    it('should have the right class names when ehiter the variant and size are not passed', () => {
        const component = renderer.create(
            <Spinner />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when a custom container class name is passed, variant is brand and size is large', () => {
        const component = renderer.create(
            <Spinner variant="brand" size="large" containerClassName="custom-container-classname" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when variant is inverse and size is medium', () => {
        const component = renderer.create(
            <Spinner variant="inverse" size="medium" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is small', () => {
        const component = renderer.create(
            <Spinner size="small" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is x-small', () => {
        const component = renderer.create(
            <Spinner size="x-small" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is xx-small', () => {
        const component = renderer.create(
            <Spinner size="xx-small" />,
        );
        expect(component).toMatchSnapshot();
    });
});
