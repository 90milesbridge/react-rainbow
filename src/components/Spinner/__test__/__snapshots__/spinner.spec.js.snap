// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> should have the right class names when a custom container class name is passed, variant is brand and size is large 1`] = `
<div
  className="slds-spinner slds-spinner_large slds-spinner_brand"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should have the right class names when ehiter the variant and size are not passed 1`] = `
<div
  className="slds-spinner slds-spinner_medium"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should have the right class names when size is small 1`] = `
<div
  className="slds-spinner slds-spinner_small"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should have the right class names when size is x-small 1`] = `
<div
  className="slds-spinner slds-spinner_x-small"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should have the right class names when size is xx-small 1`] = `
<div
  className="slds-spinner slds-spinner_xx-small"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should have the right class names when variant is inverse and size is medium 1`] = `
<div
  className="slds-spinner slds-spinner_medium slds-spinner_inverse"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;
